htsinfer/count_occurrences_motifs.py

```diff
@@ -0,0 +1,78 @@
+from Bio import SeqIO
+import numpy as np
+from tqdm import tqdm
+import pandas as pd
+import math
+
+
+def count_overlapping_motif_in_read(motif, read, overlap_cuttoff_ratio ):
+    """ Given a motif and a read, compare the motif against all substrings of the read to search for (partial) overlaps.
+    
+    Args:
+        motif: a short string, e.g., 'TTCTCCTAAGAATTGATTGAT'
+        read: a long string, whose length needs to be greater than that of the motif, e.g., 
+                    'TTCTCCTTTCTCCTAAGAATTGATTGATTCTCCTAAGAATTGATTGATAAGAATTGATTGAT'
+        overlap_cuttoff_ratio: a float between 0 and 1. Only overlaps larger than len(motif) * overlap_cuttoff_ratio will be taken into account.
+    Returns:
+        motif_overlap_count_dict: a dictionary that receives the number of partially aligned characters as input produce the number of substrings 
+             in the read that has possess this amount of partially aligned characters as output when compared to the given motif; 
+             
+        overlaps_dict: collects all partially aligned substrings for a given number of partially aligned characters.
+    """
+    len_motif = len(motif)
+
+
+    if len_motif > len(read):
+        print('read: ', read)
+        print('motif: ', motif)
+        raise ValueError('the motif length needs to be shorter than the read length!') 
```

Comment:
It would be better to put all the information about the error in the error message. Can be done by creating an error class, in which the read and motif are variables.